src/types.ts

```diff
@@ -0,0 +1,21 @@
+export interface User {
+  id: number,
+  name: string,
+  username: string,
+  email: string,
+}
+
+export interface Todo {
+  id: number,
+  title: string,
+  completed: true | false,
+  userId: number,
+}
+
+export interface PreparedTodo {
+  id: number,
+  title: string,
+  completed: true | false,
+  userId: number,
+  user: User | null;
+}
```

Comment:
These interfaces are very similar to each other, so you should use `extends` to avoid copying code